File: backend/TransportManagementSystem/Entity/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TransportManagementSystem.Entity;

public class Order
{
    [Key]
    public int Id { get; set; }

    [Required]
    public DataSource SourceSystem { get; set; } = DataSource.TMS;

    [StringLength(50)]
    public string? ExternalId { get; set; }

    [Required]
    public int CustomerId { get; set; }

    [ForeignKey("CustomerId")]
    public Customer Customer { get; set; }

    [Required]
    [StringLength(50)]
    public string Reference { get; set; } = string.Empty;

    public string? Type { get; set; } // Shampoo, Gel, Toothpaste, etc.

    [Required]
    [Column(TypeName = "decimal(10,2)")]
    public decimal Weight { get; set; } // in kg

    [Required]
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    [Required]
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedDate { get; set; }

    public string? DeliveryAddress { get; set; }

    [StringLength(500)]
    public string? Notes { get; set; }

    public int Priority { get; set; } = 5; // 1-10, 10 = highest priority

    // Navigation property for deliveries
    public ICollection<Delivery> Deliveries { get; set; } = new List<Delivery>();

    // Additional optional fields
    public string? Dimensions { get; set; } // e.g., "30x40x50 cm"
    public string? SpecialInstructions { get; set; }
}

public enum OrderStatus
{
    Pending,          // En attente
    ReadyToLoad,      // Prête au chargement
    InProgress,       // En cours de livraison
    Received,         // Réception
    Closed,           // Clôturée
    Cancelled         // Annulée
}

File: frontend/transport-management-system-web/src/app/types/order.ts
import { ICustomer } from "./customer";

export interface IOrder {
  id: number;
  customerId: number;
  customerName: string;
  customerMatricule: string;
  reference: string;
  type: string;
  weight: number;
  status: OrderStatus;
  createdDate: Date;
   deliveryDate?: Date | string;
  deliveryAddress?: string;
  notes?: string;
  priority: number;
  hasDelivery?: boolean;
  customer?: ICustomer;
  sourceSystem?: string; 
}
export interface CreateOrderDto {
  customerId: number;
  reference?: string;
  type: string;
  weight: number;
  deliveryAddress?: string;
  notes?: string;
  priority: number;
  status?: OrderStatus;
}

export interface UpdateOrderDto {
  customerId?: number;
  type?: string;
  weight?: number;
  status?: OrderStatus;
  deliveryAddress?: string;
  notes?: string;
  priority?: number;
  
}

export enum OrderStatus {
  Pending = 'pending',          // En attente
  ReadyToLoad = 'readyToLoad',  // Prête au chargement
  InProgress = 'inProgress',    // En cours de livraison
  Received = 'received',        // Réception
  Closed = 'closed',            // Clôturée
  Cancelled = 'cancelled'       // Annulée
}

export function getOrderStatusText(status: OrderStatus): string {
  switch (status) {
    case OrderStatus.Pending:
      return 'En attente';
    case OrderStatus.ReadyToLoad:
      return 'Prête au chargement';
    case OrderStatus.InProgress:
      return 'En cours de livraison';
    case OrderStatus.Received:
      return 'Réception';
    case OrderStatus.Closed:
      return 'Clôturée';
    case OrderStatus.Cancelled:
      return 'Annulée';
    default:
      return String(status);
  }
}


// Helper function to get status CSS class
export function getOrderStatusClass(status: OrderStatus): string {
  switch (status) {
    case OrderStatus.Pending:
      return 'status-pending';
    case OrderStatus.ReadyToLoad:
      return 'status-ready';
    case OrderStatus.InProgress:
      return 'status-in-progress';
    case OrderStatus.Received:
      return 'status-received';
    case OrderStatus.Closed:
      return 'status-closed';
    case OrderStatus.Cancelled:
      return 'status-cancelled';
    default:
      return '';
  }
}
